### PR TITLE
fix: Review realm objects init and enforce non nullability of primary keys

### PR DESCRIPTION
**Pull request by adrien-coye**
### Tasks

- Enforce non nullability of Primary keys
- Added missing calls to `super.init()` to respect convention.

### CODE REVIEW

File: kDriveCore/Data/Models/Drive/Drive.swift [hunk 1]
```diff
@@ -60,13 +60,13 @@ public enum MaintenanceReason: String, PersistableEnum, Codable {
...
-public class DrivePreferences: EmbeddedObject, Codable {
+public final class DrivePreferences: EmbeddedObject, Codable {
     @Persisted public var color = "#0098FF"
     @Persisted public var hide = false
 }
 
-public class Drive: Object, Codable {
-    @Persisted(primaryKey: true) public var objectId = ""
+public final class Drive: Object, Codable {
+    @Persisted(primaryKey: true) public var objectId: String = UUID().uuidString
```
Comment (PhilippeWeidmann) [inline kDriveCore/Data/Models/Drive/Drive.swift:69]: This is not correct, objectId != UUID().uuidString

Comment (adrien-coye) [reply kDriveCore/Data/Models/Drive/Drive.swift:69]: Is it less correct than the previous default that was empty string ? `""`

Comment (adrien-coye) [reply kDriveCore/Data/Models/Drive/Drive.swift:69]: I'll change this to only set a UUID in the empty init method.

Comment (PhilippeWeidmann) [reply kDriveCore/Data/Models/Drive/Drive.swift]: I wanted to ensure that we don't have places in the code where we check for .isEmpty on the objectId but this is not the case. So everything seems good 👍

File: kDriveCore/Data/Models/DriveUser.swift [hunk 2]
```diff
@@ -83,8 +83,14 @@ public class DriveUser: Object, Codable, InfomaniakUser {
...
 
+    override public init() {
+        // Required by Realm
+        super.init()
+        // Primary key set as default value
+    }
+    
     public convenience init(user: InfomaniakCore.UserProfile) {
-        self.init()
+        super.init()
```
Comment (PhilippeWeidmann) [inline kDriveCore/Data/Models/DriveUser.swift:93]: Doesn't compile

File: kDriveCore/Data/Models/Drive/Drive.swift [hunk 3]
```diff
@@ -60,13 +60,13 @@ public enum MaintenanceReason: String, PersistableEnum, Codable {
...
-public class DrivePreferences: EmbeddedObject, Codable {
+public final class DrivePreferences: EmbeddedObject, Codable {
     @Persisted public var color = "#0098FF"
     @Persisted public var hide = false
 }
 
-public class Drive: Object, Codable {
-    @Persisted(primaryKey: true) public var objectId = ""
+public final class Drive: Object, Codable {
+    @Persisted(primaryKey: true) public var objectId: String
```
Comment (PhilippeWeidmann) [inline kDriveCore/Data/Models/Drive/Drive.swift:69]: Realm doesn't seem happy if the default value isn't provided directly here

Comment (adrien-coye) [reply kDriveCore/Data/Models/Drive/Drive.swift]: Discussed and modified.

File: kDriveCore/Data/Cache/PhotoSyncSettings.swift [hunk 4]
```diff
@@ -78,11 +79,9 @@ public class PhotoSyncSettings: Object {
...
 
     override public init() {
-        // META: keep SonarCloud happy
-    }
-
-    override public class func primaryKey() -> String? {
-        return "userId"
+        // Required by Realm
+        super.init()
+        self.userId = UUID().uuidString.hashValue
```
Comment (adrien-coye) [inline kDriveCore/Data/Cache/PhotoSyncSettings.swift:84]: This can break given the way the property wrapper works.

Comment (adrien-coye) [reply kDriveCore/Data/Cache/PhotoSyncSettings.swift]: Fixed

**Pull request merged**